collections-csharp/IEnumerableVsICollectionVsIListVsList/IEnumerableVsICollectionVsIListVsList/Program.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+
+namespace IEnumerableVsICollectionVsIListVsList
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var specialCharacter = new List<char>() {'#','@','$' };
+            var arrayOfSpecialCharacter = new char[] { '=', '!' };
+
+            //IEnumerable<T>
+            var enu = new ImplementationOfIEnumerable();
+            var output = enu.CountSpecialCharacters(specialCharacter);
+            Console.WriteLine($"IEnumerable:{output}");            
+            Console.WriteLine($"IEnumerable With Yield:");
+            foreach (var num in enu.GetEvenNumberUpToTen())
+            {
+                Console.WriteLine(num);
+            }
+                  
+            //ICollection<T>
+            var icl = new ImplementationOfICollection();
+            output = icl.CountSpecialCharacters(specialCharacter);
+            Console.WriteLine($"ICollection:{output}");
+
+            //IList
+            var ils = new ImplementationOfIList();
+            output = ils.CountSpecialCharacters(specialCharacter);
+            Console.WriteLine($"IList with List input:{output}");
+            //Uncomment below line to generate runtime exception
+            //output = ils.CountSpecialCharacters(arrayOfSpecialCharacter);
```

Comment:
After that, I think this code in the main method is expendable and we can get rid of it. 
Not all the code in the article should be in the code source especailly code like showing the output. It can be temporarily written copied to the article and remove as long as different use cases are covered by the tests.

But if you think it's necessary, you can keep that, it's just a suggestion to clean the code a bit.

Reply:
I have removed the commented code and moved it to the test file.
That's a good suggestion. Just showing the output in the Main() method and the same in the test file only introduces redundancy. However, I have not removed it for now. 
Could you point out the ones I need to remove and I will do that?

Reply:
@OldPen16 we can keep it this way for this article and apply the suggestion for the next one :)